pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -690,38 +688,39 @@ func setSafeDefaults(spec *PerconaXtraDBClusterSpec, log logr.Logger) {
 // setVersion sets the API version of a PXC resource.
 // The new (semver-matching) version is determined either by the CR's API version or an API version specified via the CR's fields.
 // If the CR's API version is an empty string and last-applied-configuration from k8s is empty, it returns current operator version.
-func (cr *PerconaXtraDBCluster) setVersion() (bool, error) {
+func (cr *PerconaXtraDBCluster) setVersion() bool {
 	if len(cr.Spec.CRVersion) > 0 {
-		return false, nil
+		return false
 	}
+
 	apiVersion := version.Version
+
 	if lastCR, ok := cr.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; ok {
 		var newCR PerconaXtraDBCluster
 		err := json.Unmarshal([]byte(lastCR), &newCR)
 		if err != nil {
-			return false, errors.Wrap(err, "unmarshal cr")
-		}
-		if len(newCR.APIVersion) > 0 {
+			log.Printf("failed to unmarshal cr: %v", err)
+		} else if len(newCR.APIVersion) > 0 {
 			apiVersion = strings.Replace(strings.TrimPrefix(newCR.APIVersion, "pxc.percona.com/v"), "-", ".", -1)
 		}
 	}
 
 	cr.Spec.CRVersion = apiVersion
-	return true, nil
+	return true
 }
 
 func (cr *PerconaXtraDBCluster) Version() *v.Version {
 	return v.Must(v.NewVersion(cr.Spec.CRVersion))
 }
 
-// CompareVersionWith compares given version to current version. Returns -1, 0, or 1 if given version is smaller, equal, or larger than the current version, respectively.
-func (cr *PerconaXtraDBCluster) CompareVersionWith(version string) int {
+// CompareVersionWith compares given version to current version.
+// Returns -1, 0, or 1 if given version is smaller, equal, or larger than the current version, respectively.
+func (cr *PerconaXtraDBCluster) CompareVersionWith(ver string) int {
 	if len(cr.Spec.CRVersion) == 0 {
-		cr.setVersion()
+		_ = cr.setVersion()
```

Comment:
why do we need this empty assignment.|
Worth to put `nolint` with explanation

Reply:
There were nothing before at all. I do not think that `nolint` with explanation that we do not care about the result of this function is better than just `_` which explicitly tells us the same?

Reply:
yes, but in this case you need to go and check what is returned by function and think why it's omitted.
When the comment will explain why and what is going there